src/setup/src/setup.erl

```diff
@@ -198,9 +198,71 @@ setup_node(NewCredentials, NewBindAddress, NodeCount, Port) ->
 
 
 finish_cluster(Options) ->
+    ok = wait_connected(),
+    ok = sync_admins(),
     Dbs = proplists:get_value(ensure_dbs_exist, Options, cluster_system_dbs()),
     finish_cluster_int(Dbs, has_cluster_system_dbs(Dbs)).
 
+
+wait_connected() ->
+    Nodes = mem3:nodes() -- [node()],
+    Result = test_util:wait(fun() ->
+        case disconnected(Nodes) of
+            [] -> ok;
+            _ -> wait
+        end
+    end),
+    case Result of
+        timeout ->
+            Reason = "Cluster setup timed out waiting for nodes to connect",
+            throw({setup_error, Reason});
+        ok ->
+            ok
+    end.
+
+
+disconnected(Nodes) ->
+    lists:filter(fun(Node) ->
+        case net_adm:ping(Node) of
+            pong -> false;
+            pang -> true
+        end
+    end, Nodes).
+
+
+sync_admins() ->
+    ok = lists:foreach(fun({User, Pass}) ->
+        sync_admin(User, Pass)
+    end, config:get("admins")).
+
+
+sync_admin(User, Pass) ->
+    {Results, Errors} = rpc:multicall(nodes(), config, set,
```

Comment:
Should we use `nodes() -- [node()]`?

Reply:
The [docs](http://erlang.org/doc/man/erlang.html#nodes-0) say "Returns a list of all visible nodes in the system, except the local node. Same as nodes(visible)." so `nodes() -- [node()] == nodes()`. OTOH, we could say `mem3:nodes() -- [node()]` if you prefer?

Reply:
While testing this PR I noticed a problem. One of the nodes was not getting correct hash for `adm` user.
I added debug statement which logs the list of nodes returned by `nodes()`. Turns out that sometimes it doesn't contain all nodes of the cluster. Maybe we should use `mem3:nodes()` instead?

Reply:
@iilyak I factored a new function `other_nodes/0` which uses `mem3:nodes/0`. I subsequently re-ran the manual `dev/run` test half a dozen times, and it's working for me.